### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by Fulin-Gao** [deeplearning-wisc/opencon#2]
1. The contrastive loss is wrong. The denominator of * in log (*) should be the sum of all the representations except z.
2. For the two methods provided in Section 6 for determining the number of categories, the former is only applicable to Novel Categories Discovery, and the latter can only determine the number of categories after the whole problem is solved. Therefore, the former is not feasible and the latter is unnecessary.

**Comment by sunyiyou** [comment]
1. Please note that the problem setting is different from the traditional learning setting. There is always a reason why authors choose one strategy over another. -> Pushing away all the unlabeled data in SupCon loss easily leads to overfitting the discrimination between known vs novel. 

2. The open-world learning problem is a generalized version of NCD. I don't see why the estimation technique does not transfer. The paper's contribution also does not fall on determining the class number. We follow the baseline and measure in the same protocol.

Due to the disrespectful tone and an unlikely pleasant follow-up conversation, the issue will be closed.

**Comment by Fulin-Gao** [comment]
Thank you for your reply. We have no malice.
1. Irrelevant answer. In Supervised Contrastive Learning (https://arxiv.org/pdf/2004.11362.pdf), the Self-Supervised Contrastive Loss and Supervised Contrastive Loss made a detailed description. Instead of N(x), the denominator of the * in log(*) is the sum over all features except z.
2. If you use Han et al.'s (2019) method, they take part of the labelled data for training the network and put the other part into the unlabelled data (say there are 10 known classes, the former would use 8 classes and the remaining 2 classes would be put in later) and cluster them. In case you don't know which known classes are in the unlabelled data, ask how you should choose the remaining 2 classes. Of course, you could put all the known classes into the unlabelled sample, but then the feature extractor would focus too much on the known classes and may not be conducive to clustering afterwards.
3. If you use the other method, it is even less feasible. They use a large classification header and don't know how many known and new classes there are until after the task is completed.
Even so, I still agree with your idea of Contrastive Loss of unlabeled data, which is interesting!

**Comment by sunyiyou** [comment]
1. You did not get what a "generalized form" of N(x) is. See appendix B. 
2.3. I don't get the concern. But Cao et al. (2022) used the same as Han et al. (2019). It is not "two methods" as you said.
So one thing you can do is open another issue in https://github.com/snap-stanford/orca starting with the title "A disappointing paper" and see if they will explain it to you patiently.

**Comment by Fulin-Gao** [comment]
1. I'm sorry for the first point. It's true that your N(x)=A(x)-z. N(x) is called a negative set, which is easy to be misunderstood, because the first impression of positive and negative does not intersect.
2. In your work, what is your probe set? In addition, ORCA works without giving a desired total number of categories. 

Furthermore, I once again stated that we have no malicious intentions, and I have modified the title. I think it's normal for scientific research to diverge. Your reason for closing this conversation is thought provoking.

**Comment by sunyiyou** [comment]
You may also misunderstand the ORCA. The answer is in their paper. 

It is totally fine for divergent thinking. But please learn to be friendly if you want to start a discussion and learn sth from others.

**Comment by Fulin-Gao** [comment]
Thank you very much for your patient reply. I humbly accept your criticism and am sorry for it.

ORCA said "However, if the number of novel classes is not known ahead of time, we can initialize ORCA with a large number of prediction heads/novel classes. The ORCA objective function then infers the number of classes by not assigning any instances to unneeded prediction heads so these heads never activate."

Han et al. (2019) said "In more details, we first split the L known classes into a probe subset D^l_r of L_r classes and a training subset D^l\D^l_r containing the remaining L− L_r classes." I've read the paper carefully, and their assumption is that unlabeled samples are class disjoint from labeled samples, so their probe set is easy to choose. So how to choose the training set and the probe set in the case of joint?

**Comment by sunyiyou** [comment]
There are more details in the Exp section "Evaluation with the unknown number of novel classes." of ORCA. 

The task of estimating the class number is independent of algorithms. Therefore our paper uses the 124 class number which is the same as ORCA.
